src/components/App.tsx

```diff
@@ -107,9 +107,13 @@ import { actionFinalize } from "../actions";
 
 function withBatchedUpdates<
   TFunction extends ((event: any) => void) | (() => void)
->(func: TFunction) {
+>(
+  func: Parameters<TFunction>["length"] extends 0 | 1
+    ? TFunction
+    : "max 1 arity function allowed",
```

Comment:
`never`?

Reply:
Not sure, because then it would give this error:

```
Argument of type '(res: ActionResult, commit?: boolean) => void' is not assignable to parameter of type 'never'.ts(2345)
```

which means no one understand what's going on, so they'd have to click through `withBatchedUpdates` definition at which point they'd see:

```typescript
function withBatchedUpdates<
  TFunction extends ((event: any) => void) | (() => void)
>(func: Parameters<TFunction>["length"] extends 0 | 1 ? TFunction : never) {
  return (event => {
    unstable_batchedUpdates(func as TFunction, event);
  }) as TFunction;
}
```

and I'm pretty sure they'd not be able to immediately see what's going on either.

So I say let's keep the warning string.

Reply:
I'll merge this as is... if you still feel strongly about this, I plan to make some related fixes to history in the near future so we may revisit it there.

Reply:
Ok, I went with `never` after all, because it wouldn't show the comment explanation if the 2nd arg wasn't made optional. I added a JSDoc explanation instead. It doesn't matter, anyway.